pkg/receive/receive_test.go

```diff
@@ -5,10 +5,220 @@ package receive
+					ExternalLabels: []string{"name1=value1", "name2=value2", "name3=value3"},
+				},
+				{
+					Endpoints:      []string{"node2"},
+					Tenants:        []string{"tenant4", "tenant5", "tenant6"},
+					ExternalLabels: []string{"name4=value4", "name5=value5", "name6=value6"},
+				},
+			},
+			expectedLabelSets: []labels.Labels{
+				labels.FromStrings("replica", "test", "tenant_id", "tenant1",
+					"name1", "value1", "name2", "value2", "name3", "value3"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant2",
+					"name1", "value1", "name2", "value2", "name3", "value3"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant3",
+					"name1", "value1", "name2", "value2", "name3", "value3"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant4",
+					"name4", "value4", "name5", "value5", "name6", "value6"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant5",
+					"name4", "value4", "name5", "value5", "name6", "value6"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant6",
+					"name4", "value4", "name5", "value5", "name6", "value6"),
+			},
+		},
+		{
+			name: "Multiple hashrings - One repeated tenant",
+			cfg: []HashringConfig{
+				{
+					Endpoints:      []string{"node1"},
+					Tenants:        []string{"tenant1", "tenant2", "tenant3"},
+					ExternalLabels: []string{"name1=value1", "name2=value2", "name3=value3"},
+				},
+				{
+					Endpoints:      []string{"node2"},
+					Tenants:        []string{"tenant4", "tenant5", "tenant1"},
+					ExternalLabels: []string{"name4=value4", "name5=value5", "name6=value6"},
+				},
+			},
+			expectedLabelSets: []labels.Labels{
+				labels.FromStrings("replica", "test", "tenant_id", "tenant1",
+					"name1", "value1", "name2", "value2", "name3", "value3"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant2",
+					"name1", "value1", "name2", "value2", "name3", "value3"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant3",
+					"name1", "value1", "name2", "value2", "name3", "value3"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant4",
+					"name4", "value4", "name5", "value5", "name6", "value6"),
+				labels.FromStrings("replica", "test", "tenant_id", "tenant5",
+					"name4", "value4", "name5", "value5", "name6", "value6"),
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			var bucket objstore.Bucket
+
+			m := NewMultiTSDB(t.TempDir(), log.NewNopLogger(), prometheus.NewRegistry(),
+				&tsdb.Options{
+					MinBlockDuration:  (2 * time.Hour).Milliseconds(),
+					MaxBlockDuration:  (2 * time.Hour).Milliseconds(),
+					RetentionDuration: (6 * time.Hour).Milliseconds(),
+				},
+				labels.FromStrings("replica", "test"),
+				"tenant_id",
+				bucket,
+				false,
+				metadata.NoneFunc,
+			)
+
+			err := m.SetHashringConfig(tc.cfg)
+			require.NoError(t, err)
+
+			for _, c := range tc.cfg {
+				for _, tenantId := range c.Tenants {
+					m.TenantAppendable(tenantId)
```

Comment:
Hmm, I'm not very sure what this is doing. This seems to be continuously creating tenant appendables for every hashring in a test case?

Reply:
I thought `TenantAppendable` has to be called to make the tenants ready, like we discussed in the sync? Or I misunderstood?

Reply:
I'll add a check for the case of one tenant in multiple hashrings